pkgs/top-level/impure.nix

```diff
@@ -40,18 +40,36 @@ in
   # collections of packages.  These collection of packages are part of the
   # fix-point made by Nixpkgs.
   overlays ? let
-      dirPath = try (if pathExists <nixpkgs-overlays> then <nixpkgs-overlays> else "") "";
-      dirHome = homeDir + "/.config/nixpkgs/overlays";
-      dirCheck = dir: dir != "" && pathExists (dir + "/.");
-      overlays = dir:
-        let content = readDir dir; in
-        map (n: import (dir + ("/" + n)))
-          (builtins.filter (n: builtins.match ".*\.nix" n != null || pathExists (dir + ("/" + n + "/default.nix")))
-            (attrNames content));
+      isDir = path: pathExists (path + "/.");
```

Comment:
Is it just me, or does this not work at all if `path` is actually a path (according to `builtins.typeOf`)? For a string like `"/file.nix"`, it does indeed check `"/file.nix/."` and return `false`, but for a path like `/file.nix`, the `path + "/."` expression just returns `/file.nix` again, and `isDir` returns `true`!

This currently means I can't actually use `<nixpkgs-overlays>` pointing at a file at all, because `isDir` returns `true`, it tries to read the file as a directory, and it crashes.

I found another place in nixpkgs that does its own `isDir` in a seemingly more robust way [here](https://github.com/NixOS/nixpkgs/blob/b760158c283755e5d68a253eca0010262d0b6aa4/lib/filesystem.nix#L39-L43). It may be best to make it a top-level lib function rather than reimplementing it in multiple places.

Reply:
6 years later, I do second solsons opinion on this, that only works on strings, not actual paths.